src/Parser/PotreeBinParser.js

```diff
@@ -1,61 +1,118 @@
+        numElements: 1,
+        numByte: 2,
+        // using Float32Array because Float16Array doesn't exist
+        arrayType: Float32Array,
+        attributeName: 'intensity',
+        normalized: true,
+    },
+    CLASSIFICATION: {
+        numElements: 1,
+        arrayType: Uint8Array,
+        attributeName: 'classification',
+    },
+    // Note: at the time of writing, PotreeConverter will only generate normals in Oct16 format
+    // see PotreeConverter.cpp:121
+    // we keep all the historical value to still supports old conversion
+    NORMAL_SPHEREMAPPED: {
+        numElements: 2,
+        arrayType: Uint8Array,
+        attributeName: 'sphereMappedNormal',
+    },
+    // see https://web.archive.org/web/20150303053317/http://lgdv.cs.fau.de/get/1602
+    NORMAL_OCT16: {
+        numElements: 2,
+        arrayType: Uint8Array,
+        attributeName: 'oct16Normal',
+    },
+    NORMAL: {
+        numElements: 3,
+        arrayType: Float32Array,
+        attributeName: 'normal',
+    },
+};
+
 // Parse .bin PotreeConverter format
 export default {
     /** @module PotreeBinParser */
     /** Parse .bin PotreeConverter format and convert to THREE.Points
      * @function parse
      * @param {ArrayBuffer} buffer - the bin buffer.
+     * @param {Object} pointAttributes - the point attributes information contained in layer.metadata coming from cloud.js
      * @return {Promise} a promise that resolves with a THREE.Points.
      *
      */
-    parse: function parse(buffer) {
+    parse: function parse(buffer, pointAttributes) {
         if (!buffer) {
             throw new Error('No array buffer provided.');
         }
 
         const view = new DataView(buffer);
         // Format: X1,Y1,Z1,R1,G1,B1,A1,[...],XN,YN,ZN,RN,GN,BN,AN
-        const numPoints = Math.floor(buffer.byteLength / 16);
-
-        const positions = new Float32Array(3 * numPoints);
-        const colors = new Uint8Array(4 * numPoints);
+        let pointByteSize = 0;
+        for (const potreeName of pointAttributes) {
+            const attr = POINT_ATTTRIBUTES[potreeName];
+            pointByteSize += attr.numElements * (attr.numByte || attr.arrayType.BYTES_PER_ELEMENT);
+        }
+        const numPoints = Math.floor(buffer.byteLength / pointByteSize);
 
-        const tightbbox = new THREE.Box3();
-        tightbbox.min.set(Infinity, Infinity, Infinity);
-        tightbbox.max.set(-Infinity, -Infinity, -Infinity);
-        const tmp = new THREE.Vector3();
+        const attrs = [];
+        // get the variable attributes
+        for (const potreeName of pointAttributes) {
+            const attr = POINT_ATTTRIBUTES[potreeName];
+            attrs.push({
+                potreeName,
+                numElements: attr.numElements,
+                attributeName: attr.attributeName,
+                normalized: attr.normalized,
+                array: new attr.arrayType(attr.numElements * numPoints),
+                numByte: attr.numByte,
```

Comment:
I don't understand what you mean by that.

Reply:
here : ` numByte: attr.numByte || attr.array.BYTES_PER_ELEMENT, `
elsewhere: `attr.numByte` instead of `attr.numByte || attr.array.BYTES_PER_ELEMENT`